packages/paste-core/components/form/src/shared/FieldWrapper.tsx

```diff
@@ -26,24 +35,90 @@ const FieldWrapper: React.FC<FieldWrapperProps> = ({disabled, hasError, readOnly
     boxShadow = null;
   }
 
+  // inverse boxShadow styles
+  let InverseBoxShadow = 'shadowBorderInverse' as BoxShadow;
```

Comment:
This is starting to get a bit gnarly, the `FieldWrapper` might benefit from a button like treatment where the variants are different components we conditionally render.